minari/data_collector/data_collector.py

```diff
@@ -108,13 +108,8 @@ def __init__(
             env_spec=self.env.spec,
         )
 
-        if observation_space is None:
-            observation_space = self.env.observation_space
-        self.dataset_observation_space = observation_space
-
-        if action_space is None:
-            action_space = self.env.action_space
-        self.dataset_action_space = action_space
+        self.dataset_observation_space = self._storage.observation_space
+        self.dataset_action_space = self._storage.action_space
```

Comment:
It looks like self.dataset_observation_space is not even used; maybe we can just remove it?

Reply:
It's used in the `step` method, but true we can just get it from _storage when needed